cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -175,4 +175,124 @@ SensorBridge* MapBuilderBridge::sensor_bridge(const int trajectory_id) {
   return sensor_bridges_.at(trajectory_id).get();
 }
 
+visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodesList() {
+  visualization_msgs::MarkerArray trajectory_nodes_list;
+
+  for (int trajectory_id = 0;
+       trajectory_id < map_builder_.num_trajectory_builders();
+       ++trajectory_id) {
+    const auto trajectory_nodes =
```

Comment:
This is returning the nodes of all trajectories which you would publish multiple times depending on how many trajectories exist. You probably want to sync first as this has also changed recently. In the current implementation it will now return two nested vectors, so nodes are already separated by trajectory.